clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -8269,6 +8269,35 @@ static void addArgs(ArgStringList &DstArgs, const llvm::opt::ArgList &Alloc,
   }
 }
 
+// Partially copied from clang/lib/Frontend/CompilerInvocation.cpp
+static std::string getOptimizationLevel(const ArgList &Args) {
+  if (Arg *A = Args.getLastArg(options::OPT_O_Group)) {
+    if (A->getOption().matches(options::OPT_O0))
+      return "-O0";
+
+    if (A->getOption().matches(options::OPT_Ofast))
+      return "-O3";
+
+    assert(A->getOption().matches(options::OPT_O));
+
+    StringRef S(A->getValue());
+    if (S == "g")
+      return "-O1";
+
+    if (S == "d")
+      return "-O0";
```

Comment:
`-Og` and `-Od` are not typical Linux optimization enabling options.  Are you meaning to capture settings for Windows `clang-cl` users here?

Reply:
Yes. I added tests for `clang-cl` in `clang/test/Driver/sycl-offload.c`.